unittests/Dialect/Moore/TypesTest.cpp

```diff
@@ -68,23 +68,14 @@ TEST(TypesTest, PackedInt) {
   MLIRContext context;
   context.loadDialect<MooreDialect>();
 
-  std::tuple<IntType::Kind, StringRef, Domain, Sign> pairs[] = {
-      {IntType::Bit, "bit", Domain::TwoValued, Sign::Unsigned},
-      {IntType::Logic, "logic", Domain::FourValued, Sign::Unsigned},
-      {IntType::Reg, "reg", Domain::FourValued, Sign::Unsigned},
-      {IntType::Byte, "byte", Domain::TwoValued, Sign::Signed},
-      {IntType::ShortInt, "shortint", Domain::TwoValued, Sign::Signed},
-      {IntType::Int, "int", Domain::TwoValued, Sign::Signed},
-      {IntType::LongInt, "longint", Domain::TwoValued, Sign::Signed},
-      {IntType::Integer, "integer", Domain::FourValued, Sign::Signed},
-      {IntType::Time, "time", Domain::TwoValued, Sign::Unsigned},
-  };
-
-  for (auto pair : pairs) {
-    auto kind = std::get<0>(pair);
-    auto type = IntType::get(&context, kind);
-    ASSERT_EQ(type.getDomain(), std::get<2>(pair));
-  }
+  auto ui42 = IntType::getInt(&context, 42);
+  auto ul42 = IntType::getLogic(&context, 42);
```

Comment:
Nit: Maybe calling it `SBVType` or similar might be less confusing since there is an `int` type in SystemVerilog, and this type is not a 1-1 correspondence to that?

Reply:
Hmm yeah. SystemVerilog is very inconsistent. It calls simple bit vectors "integral" data types, or sometimes also "integers" exchangeably. And then there's the weird integer atoms which can be the `int` or `integer` you mentioned. My thinking here was that these are actually all integer data types, and the standard is just very clunky and beats around the bush with the simple bit vector types. Inside the dialect and the IR, it would be great to never have to think about the weird stuff that is in the standard, and just think of these values as integers.

Reply:
I see. In that case, the name is perfectly fine. I wish the SV standard was written less confusingly 😆

Reply:
Ugh yeah :cry: